### ISSUE TITLE
[Tool bar] Tooltips in stories Tooltips and IconOnly cannot be collapsed with `Esc` key when triggered with mouse

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3274]
### 😯 Current Behavior

When user expand tooltip with mouse cursor using `Esc` key should collapse that tooltip. This does not work right now. User can collapse tooltip with `Esc` only when tooltip is triggered by focus not hover.

https://github.com/user-attachments/assets/76dfaf6d-6f2b-4f30-a24a-4f2edf548eba

https://github.com/user-attachments/assets/01d44767-3212-4aaa-9213-4503c8f7f7f6

### 🤔 Expected Behavior/possible solution?

This should work like in Toggle group Basic story for example: 

https://github.com/user-attachments/assets/3cf8408e-9c89-4d4e-96d8-9160280a84f5

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/actions-tool-bar--tooltips or https://storybook.sanomalearning.design/?path=/story/actions-tool-bar--icon-only
2. Hover over buttons that triggers tooltips
3. When tooltip is visible click `Esc` 
4. Check if tooltip disappears after clicking `Esc`

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/actions-tool-bar--tooltips

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

v.0.0.14

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
